python/lsst/ts/tcpip/base_client_or_server.py

```diff
@@ -332,6 +333,60 @@ async def read_into(self, struct: ctypes.Structure) -> None:
             await self._close_client()
             raise
 
+    async def read_str(self) -> str:
+        """Read data ending in `TERMINATOR`.
+
+        Before returning the data, the TERMINATOR is stripped and the data is
+        decoded into an UTF-8 string.
+
+        Returns
+        -------
+        str:
+            Data decoded into a string.
+
+        Raises
+        ------
+        `ConnectionError`
+            If the connection is lost before, or while, reading.
+        `asyncio.IncompleteReadError`
+            If EOF is reached before the complete separator is found
+            and the internal buffer is reset.
+        `LimitOverrunError`
+            If the amount of data read exceeds the configured stream lmit.
+            The data is left in the internal buffer and can be read again.
```

Comment:
`UnicodeError` if decoding fails (according to the docs -- so it's what I suggest documenting -- but I actually see subclass UnicodeDecodeError on my mac).